pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/CmdProduce.java

```diff
@@ -147,6 +148,7 @@ public class CmdProduce {
     private ClientBuilder clientBuilder;
     private Authentication authentication;
     private String serviceURL;
+    private List<MessageId> messageIds = new ArrayList<>();
```

Comment:
This list will become huge when you use this tool in real world.

If you need to catch the messages you are sending you can use mockito and override the 'send' method

Reply:
Fixed